include/secp256k1.h

```diff
@@ -40,6 +40,26 @@ extern "C" {
 #  define SECP256K1_ARG_NONNULL(_x)
 # endif
 
+# if defined _MSC_VER || defined __CYGWIN__
+#  define SECP256K1_HELPER_DLL_IMPORT __declspec(dllimport)
+#  define SECP256K1_HELPER_DLL_EXPORT __declspec(dllexport)
+# else
+#  if __GNUC__ >= 4
+#   define SECP256K1_HELPER_DLL_IMPORT __attribute__ ((visibility ("default")))
+#   define SECP256K1_HELPER_DLL_EXPORT __attribute__ ((visibility ("default")))
+#  else
+#   define SECP256K1_HELPER_DLL_IMPORT
+#   define SECP256K1_HELPER_DLL_EXPORT
+#  endif
+# endif
+
+# if defined SECP256K1_STATIC
+#  define SECP256K1_API
+# elif defined SECP256K1_DLL
```

Comment:
This would be fine but it needs to be isolated from the Windows build. I'd do that by adding another helper:
```c
    define SECP256K1_HELPER_DLL_EXPORT __declspec(dllexport)
else
    if __GNUC__ >= 4
        define SECP256K1_HELPER_DLL_IMPORT __attribute__ ((visibility ("default")))
        define SECP256K1_HELPER_DLL_EXPORT __attribute__ ((visibility ("default")))
        define SECP256K1_HELPER_STATIC __attribute__ ((visibility ("hidden")))
    else
        define SECP256K1_HELPER_DLL_IMPORT
        define SECP256K1_HELPER_DLL_EXPORT
        define SECP256K1_HELPER_STATIC
    endif
endif

if defined SECP256K1_STATIC
    define SECP256K1_API SECP256K1_HELPER_STATIC
elif defined SECP256K1_DLL
```